app/models/advance_on_docket_motion.rb

```diff
@@ -32,10 +43,22 @@ def eligible_motions(person_id, appeal)
 
     def create_or_update_by_appeal(appeal, attrs)
       person_id = appeal.claimant.person.id
-      motion = eligible_due_to_appeal(appeal).for_person(person_id).first ||
-               create(person_id: person_id, appeal: appeal)
 
-      motion.update(attrs)
+      existing_motions = AdvanceOnDocketMotion.for_appeal(appeal).for_person(person_id).order(:id)
+
+      motion = for_appeal(appeal).for_person(person_id).where(reason: attrs[:reason]).order(:id).last
+      if motion
+        motion.update(attrs)
+      elsif Constants.AOD_REASONS.age == attrs[:reason]
+        # There is only one age-related reason, so if the above didn't find it, there are none.
+        create(person_id: person_id, appeal: appeal).update(attrs)
+      elsif existing_motions.any?(&:non_age_related_motion?)
```

Comment:
consider using a method alias for this to clarify what our actual check is
```suggestion
      elsif another_non_age_motion_exists
```
and then later define it
```ruby
def another_non_age_motion_exists
 existing_motions.any?(&:non_age_related_motion?)
end
```